app/migrations.go

```diff
@@ -156,3 +157,56 @@ func (a *App) DoEmojisPermissionsMigration() {
 		mlog.Critical(fmt.Sprint(result.Err))
 	}
 }
+
+func (a *App) DoGuestRolesCreationMigration() {
+	// If the migration is already marked as completed, don't do it again.
+	if result := <-a.Srv.Store.System().GetByName(GUEST_ROLES_CREATION_MIGRATION_KEY); result.Err == nil {
+		return
+	}
+
+	roles := model.MakeDefaultRoles()
+
+	allSucceeded := true
+	if result := <-a.Srv.Store.Role().GetByName(model.CHANNEL_GUEST_ROLE_ID); result.Err != nil {
+		if result := <-a.Srv.Store.Role().Save(roles[model.CHANNEL_GUEST_ROLE_ID]); result.Err != nil {
+			mlog.Critical("Failed to create new guest role to database.")
+			mlog.Critical(fmt.Sprint(result.Err))
+			allSucceeded = false
+		}
+	}
+	if result := <-a.Srv.Store.Role().GetByName(model.TEAM_GUEST_ROLE_ID); result.Err != nil {
+		if result := <-a.Srv.Store.Role().Save(roles[model.TEAM_GUEST_ROLE_ID]); result.Err != nil {
+			mlog.Critical("Failed to create new guest role to database.")
+			mlog.Critical(fmt.Sprint(result.Err))
+			allSucceeded = false
+		}
+	}
+	if result := <-a.Srv.Store.Role().GetByName(model.SYSTEM_GUEST_ROLE_ID); result.Err != nil {
+		if result := <-a.Srv.Store.Role().Save(roles[model.SYSTEM_GUEST_ROLE_ID]); result.Err != nil {
+			mlog.Critical("Failed to create new guest role to database.")
+			mlog.Critical(fmt.Sprint(result.Err))
+			allSucceeded = false
+		}
+	}
+
+	if !allSucceeded {
+		return
+	}
+
+	system := model.System{
+		Name:  GUEST_ROLES_CREATION_MIGRATION_KEY,
+		Value: "true",
+	}
+
+	if result := <-a.Srv.Store.System().Save(&system); result.Err != nil {
+		mlog.Critical("Failed to mark guest roles creation migration as completed.")
+		mlog.Critical(fmt.Sprint(result.Err))
+	}
+}
+
+func (a *App) DoAppMigrations() {
+	a.DoAdvancedPermissionsMigration()
+	a.DoEmojisPermissionsMigration()
+	a.DoGuestRolesCreationMigration()
+	a.DoPermissionsMigrations()
```

Comment:
Can you add a comment saying that this last one must always come last (i.e. any new migrations added here should be added *before* this last one)?

Reply:
Done!